File: src/main/java/pl/dolien/shop/order/OrderItem.java
package pl.dolien.shop.order;


import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;

@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class OrderItem {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private int id;

    @Column
    private String imageUrl;

    @Column
    private BigDecimal unitPrice;

    @Column
    private int quantity;

    @Column
    private Long productId;

    @ManyToOne
    @JoinColumn(name = "order_id", nullable = false)
    private Order order;
}

File: shop-ui/src/app/order/models/order-item/order-item.ts
import { CartItem } from '../../../cart/models/cart-item/cart-item';

export class OrderItem {
  name: string = '';
  imageUrl: string = '';
  quantity: number = 0;
  unitPrice: number = 0;
  productId: number = 0;

  constructor(cartItem: CartItem) {
    this.imageUrl = cartItem.imageUrl;
    this.name = cartItem.name;
    this.quantity = cartItem.quantity;
    this.unitPrice = cartItem.unitPrice;
    this.productId = cartItem.id;
  }
}
